### ISSUE TITLE
[BUG] Type mismatch in config

### ISSUE THREAD

**Issue by lonix1** [francoismassart/eslint-plugin-tailwindcss#474]
When using `// @ts-check` in a flat config file, there are errors due to missing or incorrect type definitions.

I followed the docs almost exactly. Reproduction:
```js
// @ts-check
// ...
  {
    plugins: { tailwindcss: pluginTailwind },       // <--- (1) throws ts(2322)
    extends: [pluginTailwind.configs.recommended],  // <--- (2) throws ts(2322)
    // ...
  },
```
Error 1:
```
Type '{ meta: { name: string; version: string; }; configs: { readonly recommended: Config | ConfigArray; }; rules: { "classnames-order": RuleModule<"fix:sort", [RuleOptions], unknown, RuleListener>; ... 6 more ...; "no-unnecessary-arbitrary-value": RuleModule<...>; }; }' is not assignable to type 'Plugin'.
  Types of property 'configs' are incompatible.
     ...
              Index signature for type 'string' is missing in type 'LanguageOptions'. ts(2322)

```
Error 2:
```
Type 'Config | ConfigArray' is not assignable to type 'ExtendsElement'.
  Type 'Config' is not assignable to type 'ExtendsElement'.
    ...
              Index signature for type 'string' is missing in type 'LanguageOptions'. ts(2322)
```

So the [instructions in the readme](https://github.com/francoismassart/eslint-plugin-tailwindcss#2-edit-your-eslintconfig-file) currently don't work.

I suppose the plugin should explicitly type the exported configuration array and plugin object using standard modern ESLint core types (`Linter.Config` and `Linter.Plugin`) to align with the core schema.

My `package.json` contains:
```
  "devDependencies": {
    "eslint": "10.8.0",
    "eslint-plugin-tailwindcss": "4.2.0",
    "tailwindcss": "4.3.3",
  },
```
